ansible/roles/networking/tasks/main.yaml

```diff
@@ -57,6 +57,40 @@
     - disable_iptables | default(true)
     - iptables_exists.rc == 0
 
+- name: ensure NetworkManager's conf.d directory exists
+  file:
+    path: /etc/NetworkManager/conf.d
+    state: directory
+
+# prevent NetworkManager to create conflicting entries in the routing table.
+# https://projectcalico.docs.tigera.io/maintenance/troubleshoot/troubleshooting#configure-networkmanager
+- name: configure network manager to prevent it from interfering with calico interfaces
+  copy:
+    dest: /etc/NetworkManager/conf.d/calico.conf
+    content: |
+      [keyfile]
+      unmanaged-devices=interface-name:cali*;interface-name:tunl*;interface-name:vxlan.calico;interface-name:wireguard.cali
+
+# The NetworkManager bug in RHEL 8.4 (https://bugzilla.redhat.com/show_bug.cgi?id=2007341) adds conflicting entry in route table
+# the conflicting entry hijacks traffic from host to containers.
+# Until the fix is available in upstream RHEL 8.X AMI, configure the nm-cloud-setup to prevent adding conflicting entry in route table in AWS.
+- name: configure NetworkManager-cloud-setup service to disable cloud setup on RHEL 8.x on AWS
+  when:
+    - packer_builder_type.startswith('amazon')
+    - ansible_os_family == "RedHat"
+    - ansible_distribution_major_version == '8'
+  block:
+  - name: ensure /etc/systemd/system/nm-cloud-setup.service.d exists
+    file:
+      path: /etc/systemd/system/nm-cloud-setup.service.d
+      state: directory
+  - name: create drop-in to disable nm-cloud-setup for AWS
+    copy:
+      dest: /etc/systemd/system/nm-cloud-setup.service.d/override.conf
```

Comment:
Do we not need a `daemon-reload` after?

Reply:
My understanding is that running `nm-cloud-setup` service is not needed while building the AMI itself. reloading the daemon or restarting it while building AMI should not matter since no provisioning code making node to container network call that would break. 
However, when the `override.conf` is baked into the AMI and later when this AMI is launched to create k8s node, systemctl will use the `override.conf` from beginning to skip setting up the routing table entry.

Reply:
Ahh yeah you're right, thanks for the explanation!